.github/workflows/check-links-cron.yaml

```diff
@@ -0,0 +1,69 @@
+name: Periodically Check Documentation Links
+
+on:
+  schedule:
+    - cron: '0 9 * * 1'
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - '.github/workflows/check-links-cron.yaml'
+
+permissions: read-all
+
+jobs:
+  links-checker:
+    name: Validate & Build HTML
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          persist-credentials: false 
+      - name: Build HTML
+        run: |
+          set -ex
+          # chmod +x ./Documentation/build.sh
+          # ./Documentation/build.sh
+      - name: Check Documentation links
+        run: |
+          set -ex
+          cd Documentation && sudo make html && cd ..
+
+          chmod +x ./Documentation/check-links.sh

```

Comment:
No, this should not be part of the workflow. You need to do it locally and then amend your commit to have the file with the updated permissions.

You can check locally the permissions for this file as known to Git:

```console
$ git ls-files -s Documentation/check-links.sh
100644 <SHA>
```

We want execution rights:

```console
$ chmod +x ./Documentation/check-links.sh
$ git add ./Documentation/check-links.sh
$ git commit --amend <REF>
```

Now we can check again then update the PR:

```console
$ git ls-files -s Documentation/check-links.sh
100755 <SHA>
```

Reply:
Hey, thanks for this upated it.